core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -163,6 +163,16 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
     info(s"Added fetcher to broker ${fetcherThread.sourceBroker.id} for partitions $initialOffsetAndEpochs")
   }
 
+  def addTopicIdsToFetcherThread(partitionsToUpdate: Set[(TopicPartition, Int)], topicIds: String => Option[Uuid]): Unit = {
+    lock synchronized {
+      val partitionsPerFetcher = partitionsToUpdate.groupMap(partitionsToUpdate => BrokerIdAndFetcherId(partitionsToUpdate._2, getFetcherId(partitionsToUpdate._1)))(_._1)
```

Comment:
I was having trouble setting this up, but I think if we switch to map it will be easier. Ah, It may not be supported in scala 2.12. I may need to change.